### ISSUE TITLE
Refactoring Tables

### ISSUE THREAD

**Issue by SKuipers** [GibbonEdu/core#821]
Along with our ongoing refactoring, we've been updating the HTML tables in Gibbon to use a set of PHP classes. This gives us the flexibility to add in better pagination, filtering and sorting, as well as make them mobile-responsive (soon!).

Table OOification is likely the most complicated refactoring so-far. For beginners, see #696 and #820 for a great starting point. Otherwise, give a shout with any questions, since the docs likely don't cover every use case 😅 

&nbsp;

## Info

Here's some info if you'd like to help refactor some tables in Gibbon:

👀  [Check out the ooification guide](https://gist.github.com/SKuipers/e176454a2feb555126c2147865bd0626) or the [codebase for examples](https://github.com/GibbonEdu/core/blob/v18.0.00/modules/Roll%20Groups/rollGroups.php)

👇  Pick a module that hasn't been started yet

💬  Reply here to let us know which module you're working on. We can also invite you to join the tracker spreadsheet, send an email to: support@gibbonedu.org

:octocat:  PR your changes when they're done, ideally one PR per module. [More info here](https://github.com/GibbonEdu/core/blob/master/.github/CONTRIBUTING.md#how-to-submit-a-pull-request).

&nbsp;

## Additional Modules

- [x] [Alumni](https://github.com/GibbonEdu/module-alumni) - @leofreitas
- [ ] [ATL](https://github.com/GibbonEdu/module-ATL) 
- [x] [Badges](https://github.com/GibbonEdu/module-badges) - @jimbo8098
- [x] [Credentials](https://github.com/GibbonEdu/module-credentials) - @leofreitas
- [x] Class Enrolment
- [x] Clinics
- [x] Committees
- [ ] Course Selection
- [x] Credentials
- [x] [Data Admin](https://github.com/SKuipers/module-dataAdmin) - @SKuipers 
- [x] Feed
- [x] Flexible Learning
- [x] [Free Learning](https://github.com/GibbonEdu/module-freeLearning)
- [x] [Help Desk](https://github.com/raynichc/helpdesk)
- [x] [Higher Education](https://github.com/GibbonEdu/module-higherEducation)
- [x] [House Points](https://github.com/SKuipers/gibbon-house-points) - @jimbo8098
- [ ] [IB Diploma](https://github.com/GibbonEdu/module-IBDiploma) - @powertashton
- [x] [Info Grid](https://github.com/GibbonEdu/module-infoGrid) - @jimbo8098
- [x] Mastery Transcript
- [x] Meet The Teacher
- [x] Moodle
- [x] [Policies](https://github.com/GibbonEdu/module-policies) - @leofreitas
- [x] [Query Builder](https://github.com/GibbonEdu/module-queryBuilder) - @SKuipers 
- [x] Stream
- [x] [Trip Planner](https://github.com/raynichc/Trip-Planner) - @jimbo8098

**Comment by rossdotparker** [comment]
Great work here Sandra. Hopefully this proves as well attendance as the breadcrumbs work: certainly this model of openly and clearly sharing the scope of the work and requesting  input, seems like a great way to go. Is it worth adding a link to the list of core tables that we are working on too?

**Comment by SKuipers** [comment]
Yep! Many of the core tables have varying difficulties from “seems okay” to “absolutely ridiculous” 😆 so I’m going to take a look and triage them a bit more before adding to the list.

**Comment by leofreitas** [comment]
Hi, I made my first Table OOification in module Policies (GibbonEdu/module-policies#3).

**Comment by SKuipers** [comment]
Awesome, thanks Leo! I'm going to check it out this afternoon 👍

**Comment by jimbo8098** [comment]
I'm gonna steal badges if you guys are ok with that?

**Comment by rossdotparker** [comment]
> I'm gonna steal badges if you guys are ok with that?

Jump right in @jimbo8098!

**Comment by jimbo8098** [comment]
On it! You guys really *have* been busy!

**Comment by jimbo8098** [comment]
Badges has been completed as of yesterday fyi. A PR is awaiting approval for it atm.

**Comment by jimbo8098** [comment]
Working on Info Grid now

**Comment by SKuipers** [comment]
Awesome, thanks!

**Comment by jimbo8098** [comment]
Finished info grid https://github.com/GibbonEdu/module-infoGrid/pull/6

**Comment by jimbo8098** [comment]
Gonna grab house points next.

**Comment by jimbo8098** [comment]
Must admit, kind of left this be after my last post due to other projects, however going to work on House Points today. Long time coming!

**Comment by SKuipers** [comment]
Awesome 😃 There were some unreleased changed on the House Points develop branch, so I've noted these in the changelog and pushed a v1.3.02 release. https://github.com/SKuipers/gibbon-house-points

**Comment by leofreitas** [comment]
I'm finishing Credentials. PR in next week!

**Comment by SKuipers** [comment]
> I'm finishing Credentials. PR in next week!

Brilliant, thanks!!

**Comment by jimbo8098** [comment]
If I'm not mistaken, it seems like the Help Desk module is already converted over to DataTables. I couldn't find any tables tags within that module.

**Comment by powertashton** [comment]
> @jimbo8098: If I'm not mistaken, it seems like the Help Desk module is already converted over to DataTables. I couldn't find any tables tags within that module.

@raynichc and I completely refactored the whole module, so you'd be right on that one.

**Comment by SKuipers** [comment]
Yes, well spotted, I've checked the Help Desk off on the list 😄

**Comment by jimbo8098** [comment]
FYI I've started work on the Trip Planner module's trips_manage.php page @raynichc

**Comment by leofreitas** [comment]
Hi, finished Alumni.

**Comment by SKuipers** [comment]
> Hi, finished Alumni.

Thanks Leo! 👍

**Comment by yookoala** [comment]
Questions about the remaining modules:
1. [IB Diploma](https://github.com/GibbonEdu/module-IBDiploma) module is archived. Does it still needed to be refactored?
2. Is there any link to the source code of "Course Selection"?
3. Seems all left in [ATL](https://github.com/GibbonEdu/module-ATL/) to refactor is [alt_write.php](https://github.com/GibbonEdu/module-ATL/blob/a1fa7e49e31e5ca0ffc44547699a112e0d3bfe73/ATL/atl_write.php#L267-L530) and [moduleFunctions.php](https://github.com/GibbonEdu/module-ATL/blob/a1fa7e49e31e5ca0ffc44547699a112e0d3bfe73/ATL/moduleFunctions.php#L63-L107), correct?
